packages/core/src/components/button/abstractButton.tsx

```diff
@@ -88,6 +88,13 @@ export interface IButtonProps<E extends HTMLButtonElement | HTMLAnchorElement =
      * @default "button"
      */
     type?: "submit" | "reset" | "button";
+
+    /**
+     * Provide this to give a text label to icon only buttons. You likely don't need this prop
+     * if there is other text in the button already. There is not currently a way to provide
+     * an iconTitle to `rightIcon` - please open a feature request if you find a need for this.
```

Comment:
IMO this kind of thing doesn't belong in prop documentation, it can be discussed in the github repo.

```suggestion
     * if there is other text in the button already.
```

Also, why would you have a `rightIcon` if you don't have any text?